File: LeetCode/141. Linked List Cycle/Solution.java
public class Solution {
    
    public class ListNode {
        int val;
        ListNode next;
        ListNode() {}
        ListNode(int val) { this.val = val; }
        ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    }

    public boolean hasCycle(ListNode head) {
        if(head == null || head.next==null){
            return false;
        }
        ListNode first = head ;
        ListNode last = head;
        while(first != null && first.next !=null){
            first = first.next.next;
            last = last.next;
            if (first == last ){
                return true;
            }
        }
        return false;
    }
}

File: LeetCode/141. Linked List Cycle/Solution.py
class Solution:
    def hasCycle(self, head) -> bool:
        
        if head is None:
            return False
        
        slow = fast = head
        
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
            
            if fast == slow:
                return True
            
        
        return False